gossip/gossip.go

```diff
@@ -159,7 +159,20 @@ func New(rpcContext *rpc.Context, resolvers []resolver.Resolver) *Gossip {
 	}
 
 	// Add ourselves as a SystemConfig watcher.
-	g.is.registerCallback(KeySystemConfig, g.updateSystemConfig)
+	updateC, unregister := g.RegisterUpdateChannel(KeySystemConfig)
+	g.stopper.RunWorker(func() {
+		for {
+			select {
+			case n := <-updateC:
+				g.updateSystemConfig(n)
+				updateC <- n
```

Comment:
Also, I wouldn’t necessarily say that reading and writing to the same channel is strictly unidiomatic. The pattern, while rare, is used a few times on the official website when dealing with limited access to a shared resource, such as with their semaphore and concurrent object pool examples. Still, thinking about it now, if we were to go down this route, it would probably make more sense to communicate the completion of notification processing through a method on the `Notification` struct itself (by closing a channel or something).